src/cloud-api-adaptor/podvm-mkosi/Makefile

```diff
@@ -24,21 +25,18 @@ container: fedora-binaries-builder binaries image-container
 PHONY: fedora-binaries-builder
 fedora-binaries-builder:
 	@echo "Building $(BUILDER) image..."
-ifeq ($(ARCH),s390x)
 	docker buildx build \
 		-t $(BUILDER) \
-		--build-arg ARCH=s390x \
-		--build-arg PROTOC_ARCH=s390x \
-		--build-arg YQ_ARCH=s390x \
-		--build-arg YQ_CHECKSUM=sha256:4e6324d08630e7df733894a11830412a43703682d65a76f1fc925aac08268a45 \
+		--build-arg GO_VERSION=$(GO_VERSION) \
+		--build-arg ARCH=$(ARCH) \
+		--build-arg PROTOC_VERSION=$(PROTOC_VERSION) \
+		--build-arg RUST_VERSION=$(RUST_VERSION) \
+		--build-arg YQ_VERSION=$(YQ_VERSION) \
+		--build-arg YQ_CHECKSUM=${YQ_CHECKSUM_$(ARCH)} \
```

Comment:
Why not simply use YQ_CHECKSUM? The caller will need to provide the right checksum similar to other params. Also this param name will be aligned to what is already there in the Dockerfiles

Reply:
In Makefile.defaults we have:
https://github.com/confidential-containers/cloud-api-adaptor/blob/31a32722cfd2ea5fa3173baf1b1dcfbc7949801f/src/cloud-api-adaptor/Makefile.defaults#L9-L10
So I think the two changes you've mentioned are to handle the asymmetry there between amd64 and s390x. There is maybe an argument to change that defaults to have the arch on both and then this file will be easier to understand, but that might have other consequences, so something to do later?

Reply:
Ok, so we already have arch specific checksum variables :-(
Would it be possible to reuse existing vars ie, YQ_CHECKSUM and YQ_CHECKSUM_s390x without introducing another arch variable ie YQ_CHECKSUM_amd64 ?
Or can YQ_CHECKSUM in the Makefile.defaults file be changed to YQ_CHECKSUM_amd64? (This might need changes to other files that is referring to this variable, so I'm not sure how feasible it is)

I'm not against introducing another arch specific variable; I just want us to do it consciously after ruling out other options. And also to follow a pattern which is easier to remember.

Reply:
> Or can YQ_CHECKSUM in the Makefile.defaults file be changed to YQ_CHECKSUM_amd64? (This might need changes to other files that is referring to this variable, so I'm not sure how feasible it is)

This was the change I was alluding to, but I'm concerned about the other things this might break at this point in time.

Reply:
Agree.  Let's not do this change at this moment and instead create an issue to avoid forgetting this.

Do we need the additional changes now? or only the RUST bump is good enough to proceed? 
Or we go ahead with the release since packer based image and CI is working fine and do a proper fix for mkosi post the 0.9.0 release ?

Reply:
That's a good set of questions. I think it depends if anyone is relying on the packer build for 0.9.0 release. If so we should get this merged soon (or at least the RUST VERSION hard-coded bump) and generate a new alpha pre-release. If not then maybe we can delay this and the other improvements in this PR and merge it soon after 0.9.0 is released.

Reply:
s/packer/mkosi right ?

Reply:
SInce @liudalibj already did the work needed, here is my suggestion 
1. Create a commit to align the `fedora-binaries-builder` target with `podvm-builder` target (this fixes the root cause of the mkosi issue encountered)
2. Create a commit to bump up the default version in the Dockerfile to align with versions.yaml 

If needed, `2` can be ignored for now and instead a separate PR to drop all defaults can be made to avoid these confusions.

Reply:
updated.

Reply:
did a search all other places we used `YQ_CHECKSUM` (amd64) only here we need change the value to YQ_CHECKSUM_s390x when it's s390x arch.
So we can use `YQ_CHECKSUM` directly.
If it's s390x arch then the `YQ_CHECKSUM`=`YQ_CHECKSUM_s390x`,